source/slang/slang-emit-cpp.cpp

```diff
@@ -2389,9 +2389,9 @@ void CPPSourceEmitter::emitOperandImpl(IRInst* inst, EmitOpInfo const&  outerPre
 
             if (varLayout)
             {
-                auto semanticNameSpelling = varLayout->systemValueSemantic;
-                if (semanticNameSpelling.getLength())
+                if(auto systemValueSemantic = varLayout->findSystemValueSemanticAttr())
                 {
+                    String semanticNameSpelling = systemValueSemantic->getName();
                     semanticNameSpelling = semanticNameSpelling.toLower();
 
                     if (semanticNameSpelling == "sv_dispatchthreadid")
```

Comment:
Yeah, the choice of upper-vs-lower for the CPU and GLSL back-ends isn't important - I just chose lower-case in the orignal code because it meant the string literals wouldn't be screaming at you when reading the code. We could try to consistently store upper-case in the data structures, but that gets back to the question of whether we should try to store things in a canonicalized form, or an "as the programmer wrote it" form.